#
# Disable change of the color theme (dark/light mode) in options.
#
--- a/FreeFileSync/Source/ui/gui_generated.cpp	2025-01-20 20:15:00.290738310 -0500
+++ b/FreeFileSync/Source/ui/gui_generated.cpp	2025-01-22 19:46:22.820913326 -0500
@@ -4783,6 +4791,7 @@
 	wxBoxSizer* bSizer310;
 	bSizer310 = new wxBoxSizer( wxHORIZONTAL );
 
+#if 0
 	m_bitmapColorTheme = new wxStaticBitmap( m_panel39, wxID_ANY, wxNullBitmap, wxDefaultPosition, wxDefaultSize, 0 );
 	bSizer310->Add( m_bitmapColorTheme, 0, wxALIGN_CENTER_VERTICAL|wxTOP|wxBOTTOM|wxLEFT, 5 );
 
@@ -4799,6 +4808,7 @@
 
 
 	bSizer310->Add( bSizerColorTheme, 0, wxALL|wxALIGN_CENTER_VERTICAL, 5 );
+#endif
 
 
 	bSizer186->Add( bSizer310, 0, wxEXPAND|wxALL, 5 );
@@ -5232,7 +5242,9 @@
 
 	// Connect Events
 	this->Connect( wxEVT_CLOSE_WINDOW, wxCloseEventHandler( OptionsDlgGenerated::onClose ) );
+#if 0
 	m_choiceColorTheme->Connect( wxEVT_COMMAND_CHOICE_SELECTED, wxCommandEventHandler( OptionsDlgGenerated::onChangeColorTheme ), NULL, this );
+#endif
 	m_buttonShowHiddenDialogs->Connect( wxEVT_COMMAND_BUTTON_CLICKED, wxCommandEventHandler( OptionsDlgGenerated::onShowHiddenDialogs ), NULL, this );
 	m_checkListHiddenDialogs->Connect( wxEVT_COMMAND_CHECKLISTBOX_TOGGLED, wxCommandEventHandler( OptionsDlgGenerated::onToggleHiddenDialog ), NULL, this );
 	m_bpButtonShowLogFolder->Connect( wxEVT_COMMAND_BUTTON_CLICKED, wxCommandEventHandler( OptionsDlgGenerated::onShowLogFolder ), NULL, this );
--- a/FreeFileSync/Source/ui/gui_generated.h	2025-01-20 20:15:12.549758207 -0500
+++ b/FreeFileSync/Source/ui/gui_generated.h	2025-01-22 19:48:12.706040513 -0500
@@ -1120,8 +1128,10 @@
 		wxStaticText* m_staticText93;
 		wxStaticText* m_staticText932;
 		wxStaticLine* m_staticline39;
+#if 0
 		wxStaticBitmap* m_bitmapColorTheme;
 		wxBoxSizer* bSizerColorTheme;
+#endif
 		wxStaticText* m_staticText198;
 		wxStaticLine* m_staticline191;
 		wxStaticBitmap* m_bitmapWarnings;
@@ -1184,7 +1194,9 @@
 
 		// Virtual event handlers, override them in your derived class
 		virtual void onClose( wxCloseEvent& event ) { event.Skip(); }
+#if 0
 		virtual void onChangeColorTheme( wxCommandEvent& event ) { event.Skip(); }
+#endif
 		virtual void onShowHiddenDialogs( wxCommandEvent& event ) { event.Skip(); }
 		virtual void onToggleHiddenDialog( wxCommandEvent& event ) { event.Skip(); }
 		virtual void onShowLogFolder( wxCommandEvent& event ) { event.Skip(); }
@@ -1205,7 +1217,9 @@
 
 
 	public:
+#if 0
 		wxChoice* m_choiceColorTheme;
+#endif
 		wxBitmapButton* m_bpButtonShowLogFolder;
 		fff::FolderHistoryBox* m_logFolderPath;
 		wxBitmapButton* m_bpButtonSelectAltLogFolder;
--- a/FreeFileSync/Source/ui/main_dlg.cpp	2025-01-20 20:15:31.779789648 -0500
+++ b/FreeFileSync/Source/ui/main_dlg.cpp	2025-01-22 20:30:23.468407126 -0500
@@ -6200,10 +6206,13 @@
 
 void MainDialog::onMenuOptions(wxCommandEvent& event)
 {
+#if 0
     const ColorTheme colorThemeOld = globalCfg_.appColorTheme;
+#endif
 
     showOptionsDlg(this, globalCfg_);
 
+#if 0
     if (!equalAppearance(globalCfg_.appColorTheme, colorThemeOld))
     {
         if (!folderCmp_.empty()) //otherwise: why bother the user?
@@ -6250,6 +6259,7 @@
             catch (const FileError& e) { showNotificationDialog(this, DialogInfoType::error, PopupDialogCfg().setDetailInstructions(e.toString())); }
         }
     }
+#endif
 }
 
 
--- a/FreeFileSync/Source/ui/small_dlgs.cpp	2025-01-20 20:31:14.579742266 -0500
+++ b/FreeFileSync/Source/ui/small_dlgs.cpp	2025-01-22 20:33:48.177367133 -0500
@@ -1294,7 +1294,9 @@
     void selectSound(wxTextCtrl& txtCtrl);
 
     void onChangeSoundFilePath(wxCommandEvent& event) override { updateGui(); }
+#if 0
     void onChangeColorTheme   (wxCommandEvent& event) override { updateGui(); }
+#endif
 
     void onPlayCompareDone (wxCommandEvent& event) override { playSoundWithDiagnostics(trimCpy(m_textCtrlSoundPathCompareDone ->GetValue())); }
     void onPlaySyncDone    (wxCommandEvent& event) override { playSoundWithDiagnostics(trimCpy(m_textCtrlSoundPathSyncDone    ->GetValue())); }
@@ -1326,6 +1328,7 @@
 
     const GlobalConfig defaultCfg_;
 
+#if 0
     EnumDescrList<ColorTheme> enumColorTheme_
     {
         *m_choiceColorTheme,
@@ -1337,6 +1340,7 @@
     };
 
     std::optional<ColorTheme> colorThemeIcon_; //perf: don't update icon unless needed
+#endif
 
     std::vector<std::tuple<std::function<bool(const GlobalConfig& cfg)> /*get dialog shown status*/,
         std::function<void(GlobalConfig& gs, bool show)> /*set dialog shown status*/,
@@ -1482,8 +1486,10 @@
     m_checkBoxCopyLocked     ->SetValue(globalCfg.copyLockedFiles);
     m_checkBoxCopyPermissions->SetValue(globalCfg.copyFilePermissions);
 
+#if 0
     bSizerColorTheme->Show(darkModeAvailable());
     enumColorTheme_.set(globalCfg.appColorTheme);
+#endif
 
     m_checkBoxLogFilesMaxAge->SetValue(globalCfg.logfilesMaxAgeDays > 0);
     m_spinCtrlLogFilesMaxAge->SetValue(globalCfg.logfilesMaxAgeDays > 0 ? globalCfg.logfilesMaxAgeDays : GlobalConfig().logfilesMaxAgeDays);
@@ -1685,6 +1691,7 @@
 
 void OptionsDlg::updateGui()
 {
+#if 0
     if (!colorThemeIcon_ || *colorThemeIcon_ != enumColorTheme_.get())
         switch (enumColorTheme_.get())
         {
@@ -1699,6 +1706,7 @@
                 break;
         }
     colorThemeIcon_ = enumColorTheme_.get();
+#endif
 
     m_spinCtrlLogFilesMaxAge->Enable(m_checkBoxLogFilesMaxAge->GetValue());
 
@@ -1771,7 +1779,9 @@
     m_checkBoxCopyLocked     ->SetValue(defaultCfg_.copyLockedFiles);
     m_checkBoxCopyPermissions->SetValue(defaultCfg_.copyFilePermissions);
 
+#if 0
     enumColorTheme_.set(defaultCfg_.appColorTheme);
+#endif
 
     unsigned int itemPos = 0;
     for (const auto& [dlgShown, dlgSetShown, msg] : hiddenDialogCfgMapping_)
@@ -1815,7 +1825,9 @@
     globalCfgOut_.copyLockedFiles     = m_checkBoxCopyLocked->GetValue();
     globalCfgOut_.copyFilePermissions = m_checkBoxCopyPermissions->GetValue();
 
+#if 0
     globalCfgOut_.appColorTheme = enumColorTheme_.get();
+#endif
 
     globalCfgOut_.logFolderPhrase = logFolderPhrase;
     m_logFolderPath->getHistory()->addItem(logFolderPhrase);
--- a/FreeFileSync/Source/config.cpp	2025-01-22 20:38:40.656362540 -0500
+++ b/FreeFileSync/Source/config.cpp	2025-01-22 20:39:06.258364213 -0500
@@ -100,6 +100,7 @@
 }
 
 
+#if 0
 template <> inline
 void writeText(const ColorTheme& value, std::string& output)
 {
@@ -131,6 +132,7 @@
         return false;
     return true;
 }
+#endif
 
 
 template <> inline
@@ -1379,7 +1381,9 @@
     else
         in2["Language"].attribute("Code", cfg.programLanguage);
 
+#if 0
     in2["ColorTheme"].attribute("Appearance", cfg.appColorTheme);
+#endif
 
     in2["FailSafeFileCopy"         ].attribute("Enabled", cfg.failSafeFileCopy);
     in2["CopyLockedFiles"          ].attribute("Enabled", cfg.copyLockedFiles);
@@ -2139,7 +2143,9 @@
 void writeConfig(const GlobalConfig& cfg, XmlOut& out)
 {
     out["Language"].attribute("Code", cfg.programLanguage);
+#if 0
     out["ColorTheme"].attribute("Appearance", cfg.appColorTheme);
+#endif
 
     out["FailSafeFileCopy"         ].attribute("Enabled", cfg.failSafeFileCopy);
     out["CopyLockedFiles"          ].attribute("Enabled", cfg.copyLockedFiles);
